src/RegistrationMsg.chpl

```diff
@@ -96,6 +99,199 @@ module RegistrationMsg
+        }
+
+        repMsg = "categorical;created %s".format(cats);
+        // Check if the categories is numeric or string, if string add byte size
+        if (isStringAttrib(cats)) {
+            var s = getSegString("%s.categories".format(name), st);
+            repMsg += "+created bytes.size %t".format(s.nBytes);
+        }
+
+        repMsg += ";created %s".format(codes);
+
+        // Optional components of categorical
+        if st.contains("%s.permutation".format(name)) {
+            var perm = st.attrib("%s.permutation".format(name));
+            if (perm.startsWith("Error:")) { 
+                var errorMsg = perm;
+                regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR); 
+            }
+            repMsg += ";created %s".format(perm);
+        }
+        if st.contains("%s.segments".format(name)) {
+            var segs = st.attrib("%s.segments".format(name));
+            if (segs.startsWith("Error:")) { 
+                var errorMsg = segs;
+                regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR); 
+            }
+            repMsg += ";created %s".format(segs);
+        }
+
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+
+    /* 
+    Compile the component parts of a SegArray attach message 
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg name: name of SymTab element
+    :type name: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc attachSegArrayMsg(cmd: string, name: string, st: borrowed SymTab): MsgTuple throws {
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting SegArray components for '%s'".format(cmd, name));
+
+        var repMsg: string;
+
+        var segs = st.attrib("%s_segments".format(name));
+        if (segs.startsWith("Error:")) { 
+            var errorMsg = segs;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        var vals = st.attrib("%s_values".format(name));
+        if (vals.startsWith("Error:")) { 
+            var errorMsg = vals;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        var lens = st.attrib("%s_lengths".format(name));
+        if (lens.startsWith("Error:")) { 
+            var errorMsg = lens;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        repMsg = "segarray;created %s;created %s;created %s".format(segs, vals, lens);
```

Comment:
replace ; with +

Reply:
Updated to use `+` delimited string in the response message